### PR TITLE
[IOCOM-686] Add message generation with remote content

### PR DESCRIPTION
**Pull request by adelloste**
## Short description
This PR allows to generate messages with remote content.

## List of changes proposed in this pull request
- `src/config`: added configuration for messages with remote content
- `src/populate-persistence.ts`: added generation of messages with remote content (saved in-memory)
- `src/features/messages/persistence/messagesPayload.ts`: added `withRemoteContent` function that adds remote content to the message

## How to test
Run the server instance with different configurations and check that the payload returned by the `/messages/:id` and `/third-party-messages/:id` endpoints is correct

### CODE REVIEW

File: src/features/messages/persistence/messagesPayload.ts [hunk 1]
```diff
@@ -61,21 +63,33 @@ export const withDueDate = (
...
-  attachmentCount: number
+  markdown: string
 ): ThirdPartyMessageWithContent => ({
   ...message,
   content: {
     ...message.content,
     third_party_data: {
       ...message.content.third_party_data,
-      id: message.id as NonEmptyString,
-      has_attachments: true
```
Comment (Vangaorth) [inline src/features/messages/persistence/messagesPayload.ts:74]: This flag is sent by the real backend, we should keep it and assign a proper value to it. 
We also need to add the `has_remote_content: boolean` flag.
What do you think?

File: src/features/messages/services/messagesService.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ const getPublicMessages = (
       B.fold(
         () => ({}),
         () => {
-          const { content, is_read, is_archived, has_attachments } =
```
Comment (Vangaorth) [inline src/features/messages/services/messagesService.ts:86]: See previous comment on keeping this value, also with the new `has_remote_content`

File: src/features/messages/services/messagesService.ts [hunk 3]
```diff
@@ -93,7 +93,6 @@ const getPublicMessages = (
             category: getMessageCategory(message),
             is_read,
             is_archived,
-            has_attachments,
```
Comment (Vangaorth) [inline src/features/messages/services/messagesService.ts:96]: See previous comment on keeping this value, also with the new `has_remote_content`

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: src/features/messages/persistence/messagesPayload.ts (hunk 1 above)
Comment (adelloste) [reply src/features/messages/persistence/messagesPayload.ts:74]: Restored `has_attachments` flag

**Review by Vangaorth — APPROVED** [review]
LGTM

**Pull request merged**